From d6bd546d06bcc25710fc01f32a27029b2ad4d0b7 Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@nxp.com>
Date: Mon, 19 Nov 2018 17:34:44 +0800
Subject: [PATCH 5161/5242] MLK-20207-2: input: synaptics_dsx_i2c: fix
 coverity Unsigned compared against 0 issue.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

commit  fd6faa8221f25d2f5a91507d2ab993888f97c8a6 from
https://source.codeaurora.org/external/imx/linux-imx.git

CID 3869705: Unsigned compared against 0 (NO_EFFECT)
unsigned_compare: This greater-than-or-equal-to-zero comparison of an
unsigned value is always true. attr_count >= 0.

Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
(cherry picked from commit 7380600ba0d0763b6081c07f5ddcb018a54314ca)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../touchscreen/synaptics_dsx/synaptics_dsx_i2c.c  |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/synaptics_dsx/synaptics_dsx_i2c.c b/drivers/input/touchscreen/synaptics_dsx/synaptics_dsx_i2c.c
index 64089d3..7194122 100644
--- a/drivers/input/touchscreen/synaptics_dsx/synaptics_dsx_i2c.c
+++ b/drivers/input/touchscreen/synaptics_dsx/synaptics_dsx_i2c.c
@@ -3162,7 +3162,7 @@ static int synaptics_rmi4_probe(struct i2c_client *client,
 		const struct i2c_device_id *dev_id)
 {
 	int retval, ret;
-	unsigned char attr_count;
+	signed char attr_count;
 	struct synaptics_rmi4_data *rmi4_data;
 	struct device_node *np = client->dev.of_node;
 #if 0
-- 
1.7.9.5

